bouncer/static/scripts/redirect.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+/** The ID of the Chrome extension we want to talk to. */
+var EDITOR_EXTENSION_ID = 'oldbkmekfdjiffgkconlamcngmkioffd';
+
+/** Return the settings object that the server injected into the page. */
+function getSettings(document) {
+  return JSON.parse(
+    document.querySelector('script.js-bouncer-settings').textContent);
+}
+
+/** Navigate the browser to the given URL. */
+function navigateTo(url) {
+  window.location = url;
+}
+
+/** Navigate the browser to the requested annotation.
+ *
+ * If the browser is Chrome and our Chrome extension is installed then
+ * navigate to the annotation's direct link for the Chrome extension.
+ * If the Chrome extension isn't installed or the browser isn't Chrome then
+ * navigate to the annotation's Via direct link.
+ *
+ */
+function redirect(navigateTo_) {
+  // Use the test navigateTo() function if one was passed in, the real
+  // navigateTo() otherwise.
+  navigateTo = navigateTo_ || navigateTo;
+
+  var settings = getSettings(document);
+
+  if (window.chrome) {
+    // The user is using Chrome, redirect them to our Chrome extension if they
+    // have it installed, via otherwise.
+    chrome.runtime.sendMessage(
+      EDITOR_EXTENSION_ID,
+      'Hi?',
```

Comment:
I suggest it might be a good idea to exchange structured objects with the extension so that you have room to add additional information if you need in future. The pattern I've been following is that all messages sent back and forth between scripts are objects with a 'type' property which is a string identifying the message, plus additional info as needed.